### PR TITLE
feat: Implement the new top bar component

### PR DESCRIPTION
**Pull request by mkarajohn**
## Description

Resolves [DSTRBTN-540]

## Test Plan

[DSTRBTN-540]: https://orfium.atlassian.net/browse/DSTRBTN-540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkarajohn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #74** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/74" target="_blank">Graphite</a> 
    * **PR #75** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/75" target="_blank">Graphite</a> 
      * **PR #76** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/76" target="_blank">Graphite</a> 
        * **PR #77** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/77" target="_blank">Graphite</a> 
          * **PR #78** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/78" target="_blank">Graphite</a> 
            * **PR #79** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/79" target="_blank">Graphite</a> 
              * **PR #80** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/80" target="_blank">Graphite</a> 
                * **PR #81** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/81" target="_blank">Graphite</a> 
                  * **PR #82** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/82" target="_blank">Graphite</a> 
                    * **PR #83** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/83" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/83?utm_source=stack-comment).

File: src/ui/TopBar/components/UserMenu/UserMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            if (!open) {
+              setOpen(true);
+            }
+          }}
+          onKeyDown={(e) => {
+            if ((e.code === 'Enter' || e.code === 'Space') && !open) {
+              setOpen(true);
+            }
+          }}
+          className={open ? '' : 'hidden'}
```
Comment (panvourtsis) [inline src/ui/TopBar/components/UserMenu/UserMenu.tsx:89]: kudos on accessibility 🎊  also as you added accessibility parts aria-hidden here could be good

**Review by panvourtsis — APPROVED** [review]
nicely done 🚀

File: src/ui/TopBar/components/UserMenu/UserMenu.tsx (hunk 1 above)
Comment (mkarajohn) [reply src/ui/TopBar/components/UserMenu/UserMenu.tsx]: Resolved in latest version

**Review by panvourtsis — APPROVED** [review]

**Comment by mkarajohn** [comment]
### Merge activity

* **Oct 24, 10:43 AM**: @mkarajohn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/83).

**Pull request merged**